instrumentation/src/main/java/io/opentelemetry/android/features/diskbuffering/SignalDiskExporter.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.features.diskbuffering
+
+import androidx.annotation.WorkerThread
+import io.opentelemetry.contrib.disk.buffering.LogRecordDiskExporter
+import io.opentelemetry.contrib.disk.buffering.MetricDiskExporter
+import io.opentelemetry.contrib.disk.buffering.SpanDiskExporter
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+
+/**
+ * Entrypoint to read and export previously cached signals.
+ */
+class SignalDiskExporter internal constructor(
```

Comment:
Something is bugging me about the name of this class -- it makes the reader think that this is responsible for exporting a single signal to disk...but really, it's just a container (not really an entrypoint(?)) for 3 type-specific `@WorkerThread` methods that read a batch from disk and export to a signal type-specific delegate.

I think part of the problem might stem from something in the upstream contrib repo, so I think it's probably fine to continue with this approach until we can address it there. However, I think we should rename this...and of course naming is hard.

How about something like `StoredBatchReadingExporters` or `BatchExportingDiskReaders` or similar?

Reply:
> I think part of the problem might stem from something in the upstream contrib repo, so I think it's probably fine to continue with this approach until we can address it there. However, I think we should rename this...and of course naming is hard.

After taking a look at [your PR](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1161) I think we definitely can simplify this further once that's merged! In the meantime, yeah the name isn't ideal, it's always a struggle for me to come up with good names X_x so thanks for the suggestions. This object is really just a utility to trigger "`FromDisk`-exports", if that term makes sense, of all types of signals, so it's supposed to be kinda handy to avoid having to manually keeping track of all the `ToDisk`-exporters, which are set internally during the RUM initialization, so it could be awkward to get them from there, so instead this utility serves as a container for them all, as well as providing `FromDisk`-exporting functions in a singleton that's accessible from anywhere, even from custom scheduler impls.

Based on that, maybe a good name could be something like `StoredFromDiskExporters` or maybe `FromDiskExporters`, which would make sense after [your PR](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1161) gets merged, though I guess it depends on how quick we need to merge this one (I wouldn't mind waiting for the contrib one).

Reply:
Yeah let's definitely punt on this until the upstream changes land here. 🤘🏻 Totally fine with it being this way for now.